workflow/executor/resource.go

```diff
@@ -165,8 +169,8 @@ func (we *WorkflowExecutor) WaitResource(resourceNamespace string, resourceName
 	}
 
 	// Start the condition result reader using PollImmediateInfinite
-	// Poll intervall of 5 seconds serves as a backoff intervall in case of immediate result reader failure
-	err := wait.PollImmediateInfinite(time.Second*5,
+	// Poll interval serves as a backoff interval in case of immediate result reader failure
+	err := wait.PollImmediateInfinite(resourceStateCheckInterval,
```

Comment:
This change is not really necessary and will have no affect on performance. 

`checkResourceState` is actually a long, blocking watch on the resource and not a not a get. So the only time the interval comes into play is when the watch fails, which should be rare.

Reply:
Got it. I previously missed the `-w` part and thought it was a get. So looks like there should be no performance issue here and https://github.com/argoproj/argo/issues/4467 is no longer necessary. Am I right?

Reply:
Correct. I think we should close this.